src/services/github.service.ts

```diff
@@ -90,24 +85,27 @@ export class GithubService {
         map((repos) => repos.filter((repo) => !repo.fork)),
         map((repos) =>
           repos.map((repo) => {
-            return {
-              repository_name: repo.name,
-              owner_login: repo.owner.login,
-              branches: [],
-            };
+            return forkJoin({
```

Comment:
Creating this two observables does not bring any value.

Create a method that will do it for a single repo:

`
private getRepoWithBranches(user, repo): Observable<Repository> {

return this.getBranches(user, repo.name, headers)
.pipe(
(map(branches: Branch[]) => {
return {
repository_name: repo.name,
owner_login: repo.owner.login,  // that is also a bit unsafe - is there any place where You validate GitHub response if that field exists?
branches
}
}
)

}
`

and then from Your current `getNotForkRepos` call it for every repo

Reply:
In general - don't create Observables, Promises or any other complex wrappers if You don't need to - it does come for free, it does bring an overhead.

Reply:
I just tried to follow your previous comment _"one function to getBranches and then other to getRepos and from getRepos You cal getBranches"_. And the same examples of implementation I found in **meet-the-team repo**.  :))
![Untitled](https://user-images.githubusercontent.com/91031463/140275030-285b4a51-29b0-4469-83f0-baa5e0b66c48.png)

Reply:
1. That was a simplification, that logic should be separated and SRP should be followed. 
2. Comparing Your code to mtt  - You have 3 observables, 2 are redundant - I think You know the answer why this is wrong

Reply:
Yes sure, I will fix it. thanks

Reply:
I fixed it, check please when You have time